src/H5encode.h

```diff
@@ -224,7 +224,7 @@
         n = 0;                                                                                               \
         (p) += 8;                                                                                            \
         for (_i = 0; _i < sizeof(uint64_t); _i++)                                                            \
-            n = (n << 8) | *(--p);                                                                           \
+            n = (uint64_t)(((uint64_t)n << 8) | *(--p));                                                     \
```

Comment:
Casting this back to a `uint64_t` seems to be showing a warning:

```
conversion from ‘long unsigned int’ to ‘uint32_t’ {aka ‘unsigned int’} may change value [-Warith-conversion]
  227 |             n = (uint64_t)(((uint64_t)n << 8) | *(--p));                                                     \
      |                 ^
src/H5Olayout.c:398:33: note: in expansion of macro ‘H5F_DECODE_LENGTH’
  398 |                                 H5F_DECODE_LENGTH(f, p, mesg->storage.u.chunk.u.single.nbytes);
      |                                 ^~~~~~~~~~~~~~~~~
```

Seems like we could end up trying to UINT64DECODE into a `uint32_t` buffer

Reply:
Updated this to decode into a uint64_t and then checked_assign it to the uint32_t field.